File: 125B/main.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
    string xml;
    getline(cin, xml);
    
    int indent = 0;
    int i = 0;
    int n = xml.length();
    vector<string> result;
    
    while (i < n) {
        if (xml[i] == '<') {
            if (i + 1 < n && xml[i+1] == '/') {
                indent--;
                int tag_start = i;
                while (i < n && xml[i] != '>') {
                    i++;
                }
                string tag = xml.substr(tag_start, i - tag_start + 1);
                result.push_back(string(2 * indent, ' ') + tag);
            } else {
                int tag_start = i;
                while (i < n && xml[i] != '>') {
                    i++;
                }
                string tag = xml.substr(tag_start, i - tag_start + 1);
                result.push_back(string(2 * indent, ' ') + tag);
                indent++;
            }
            i++;
        } else {
            i++;
        }
    }
    
    for (const string& line : result) {
        cout << line << endl;
    }
    
    return 0;
}

File: 125B/main.py
xml = input().strip()
indent = 0
i = 0
n = len(xml)
result = []

while i < n:
    if xml[i] == '<':
        if xml[i+1] == '/':
            indent -= 1
            tag_start = i
            while xml[i] != '>':
                i += 1
            tag = xml[tag_start:i+1]
            result.append('  ' * indent + tag)
        else:
            tag_start = i
            while xml[i] != '>':
                i += 1
            tag = xml[tag_start:i+1]
            result.append('  ' * indent + tag)
            indent += 1
        i += 1
    else:
        i += 1

print('\n'.join(result))